packages/salesforcedx-vscode-soql/src/telemetry/index.ts

```diff
@@ -4,11 +4,11 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-import { TelemetryService } from '@salesforce/salesforcedx-utils-vscode';
+import { TelemetryProvider } from '@salesforce/salesforcedx-utils-vscode';
 import { JsonMap } from '@salesforce/ts-types';
 import * as vscode from 'vscode';
 
-export const telemetryService = TelemetryService.getInstance();
+export const telemetryService = TelemetryProvider.getInstance();
```

Comment:
dont all these other non core extensions need to pass the right extension name to TelemetryProvider.getInstance() now? IOW dont these other extensions also have the same issue as the egpt extension?

Reply:
@gbockus-sf do you have idea about this? I think that is about the definition of "core". Does "core" refer to all extensions in the mono repo or just "salesforce-vscode-core"? My guts is that the issue is only caused by external extensions so far. So I keep the same instance of the internal extensions.

Reply:
They don't have the same issue b/c they aren't sharing an _instance_ of the telemetry service.  This is unique to the use of the service via the API.